bindings/wasm/tests/jwk_storage.ts

```diff
@@ -0,0 +1,199 @@
+            alg: JwsAlgorithm.EdDSA,
+            key_ops: [JwkOperation.Sign, JwkOperation.Verify],
+            crv: EdCurve.Ed25519,
+            d: "nWGxne_9WmC6hEr0kuwsxERJxWl7MmkZcDusAxyuf2A",
+            x: "11qYAYKxCrfVS_7TyWQHOg7hcvPapiMlrwIaaPcHURo",
+        };
+
+        const localJwk = new Jwk(jwkParams);
+        assert.ok(await memstore.insert(localJwk));
+
+        const pubLocalJwk = new Jwk({
+            ...jwkParams,
+            // Null out the private key component
+            d: undefined,
+        });
+
+        // INVALID: Inserting a JWK without the private key component should fail.
+        assert.rejects(async () => {
+            await memstore.insert(pubLocalJwk);
+        });
+    });
+});
+
+export class JwkMemStore implements JwkStorage {
+    /** The map from key identifiers to Jwks. */
+    private _keys: Map<string, Jwk>;
+
+    /** Creates a new, empty `MemStore` instance. */
+    constructor() {
+        this._keys = new Map();
+    }
+
+    public static ed25519KeyType(): string {
+        return "Ed25519";
+    }
+
+    public async generate(keyType: string, algorithm: JwsAlgorithm): Promise<JwkGenOutput> {
+        if (keyType !== JwkMemStore.ed25519KeyType()) {
+            throw new Error(`unsupported key type ${keyType}`);
+        }
+
+        if (algorithm !== JwsAlgorithm.EdDSA) {
+            throw new Error(`unsupported algorithm`);
+        }
+
+        const keyId = randomKeyId();
+        const keyPair = new KeyPair(KeyType.Ed25519);
+
+        const jwk = encodeJwk(keyPair, algorithm);
+
+        this._keys.set(keyId, jwk);
+
+        return new JwkGenOutput(keyId, jwk.toPublic());
+    }
+
+    public async sign(keyId: string, data: Uint8Array, publicKey: Jwk): Promise<Uint8Array> {
+        if (publicKey.alg() !== JwsAlgorithm.EdDSA) {
+            throw new Error("unsupported JWS algorithm");
+        } else {
+            if (publicKey.paramsOkp()?.crv !== (EdCurve.Ed25519 as string)) {
+                throw new Error("unsupported Okp parameter");
+            }
+        }
+
+        const jwk = this._keys.get(keyId);
+
+        if (jwk) {
+            const keyPair = decodeJwk(jwk);
+            return Ed25519.sign(data, keyPair.private());
+        } else {
+            throw new Error(`key with id ${keyId} not found`);
+        }
+    }
+
+    public async insert(jwk: Jwk): Promise<string> {
+        const keyId = randomKeyId();
+
+        if (!jwk.isPrivate) {
+            throw new Error("expected a JWK with all private key components set");
+        }
+
+        if (!jwk.alg()) {
+            throw new Error("expected a Jwk with an `alg` parameter");
+        }
+
+        this._keys.set(keyId, jwk);
+
+        return keyId;
+    }
+
+    public async public(keyId: string): Promise<Jwk> {
+        const jwk = this._keys.get(keyId);
+
+        if (jwk) {
+            return jwk.toPublic();
+        } else {
+            throw new Error("key not found");
+        }
+    }
```

Comment:
We removed the `public` method from the interface, can we double-check if we need this function here?